archiveFactSheets/archive.py

```diff
@@ -1,45 +1,138 @@
+LEANIX_SUBDOMAIN = os.getenv('LEANIX_SUBDOMAIN')
 
+LEANIX_AUTH_URL = f'https://{LEANIX_SUBDOMAIN}.leanix.net/services/mtm/v1/oauth2/token' 
+LEANIX_REQUEST_URL = f'https://{LEANIX_SUBDOMAIN}.leanix.net/services/pathfinder/v1/graphql'
+
+IMPORT_FILE = os.getenv('IMPORT_FILE')
+
+
+#LOGIC
 # Get the bearer token - see https://dev.leanix.net/v4.0/docs/authentication
-response = requests.post(auth_url, auth=('apitoken', api_token),
-                         data={'grant_type': 'client_credentials'})
-response.raise_for_status() 
-access_token = response.json()['access_token']
-auth_header = 'Bearer ' + access_token
-header = {'Authorization': auth_header}
+def get_bearer_token(auth_url, api_token):
+    """Function to retrieve the bearer token for authentication
+
+    Args:
+        auth_url (str): URL to retrieve the bearer token from
+        api_token (str): The api-token to authenticate with
+
+    Returns:
+        dict: Dictionary containing the bearer token
+    """
+    if not LEANIX_API_TOKEN:
+        raise Exception('A valid token is required')
+    response = requests.post(auth_url, auth=('apitoken', api_token),
+                             data={'grant_type': 'client_credentials'},
+                             timeout=TIMEOUT)
+    response.raise_for_status() 
+    access_token = response.json()['access_token']
+    auth_header = 'Bearer ' + access_token
+    header = {'Authorization': auth_header}
+    return header
+
 
 # General function to call GraphQL given a query
-def call(query):
-  data = {"query" : query}
-  json_data = json.dumps(data)
-  response = requests.post(url=request_url, headers=header, data=json_data)
-  response.raise_for_status()
-  return response.json()
+def call(query, header, request_url):
+    """Function that allows the user to perform graphql queries.
+
+    Args:
+        query (str): Query the user wants to perform on his workspace.
+
+    Returns:
+        str: JSON response string for the given query.
+    """
+    data = {"query" : query}
+    json_data = json.dumps(data)
+    response = requests.post(url=request_url, headers=header, data=json_data, timeout=TIMEOUT)
+    response.raise_for_status()
+    return response.json()
+
 
 # Delete the subscription
-def archiveFactSheets(id):
-  query = """
-mutation {
+def archiveFactSheets(id, header):
+    """Function to construct the query for archiving a certain factsheet.
+
+    Args:
+        id (str): ID for the factsheets that is to be archived.
+    """  
+    query = """
+    mutation {
       updateFactSheet(id: "%s", comment: "Archive", patches: {op: replace, path: \"/status\", value: \"ARCHIVED\"}, validateOnly: false) {
         factSheet {
           id
         }
       }
     }
-  """ % (id)
-  print ("delete " + id)
-  response = call(query)
-  print (response)
+    """ % (id)
+    logging.info("delete " + id)
+    response = call(query, header, LEANIX_REQUEST_URL)
+    logging.info(response)
```

Comment:
It is better to use `debug` for responses as those usually are not meant to be logged (expensive and noisy).
```suggestion
    logging.debug(response)
```